internal/provider/resources/resource_deployment_test.go

```diff
@@ -324,7 +325,228 @@ func TestAcc_ResourceDeploymentStandard(t *testing.T) {
+					Name:              scalingSpecDeploymentName,
+					Description:       utils.TestResourceDescription,
+					Region:            "us-east4",
+					CloudProvider:     "GCP",
+					Executor:          "CELERY",
+					SchedulerSize:     "SMALL",
+					IsDevelopmentMode: true,
+					ScalingSpec: `
+						scaling_spec = {
+							hibernation_spec = {
+								schedules = [{
+								  hibernate_at_cron    = "1 * * * *"
+								  is_enabled           = true
+								  wake_at_cron         = "59 * * * *"
+								}]
+							}
+						}`}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(scalingSpecResourceVar, "scaling_spec.hibernation_spec.schedules.0.hibernate_at_cron", "1 * * * *"),
+					resource.TestCheckResourceAttr(scalingSpecResourceVar, "scaling_spec.hibernation_spec.schedules.0.is_enabled", "true"),
+					resource.TestCheckResourceAttr(scalingSpecResourceVar, "scaling_spec.hibernation_spec.schedules.0.wake_at_cron", "59 * * * *"),
+					resource.TestCheckNoResourceAttr(scalingSpecResourceVar, "scaling_spec.hibernation_spec.override"),
+				),
+			},
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + standardDeployment(standardDeploymentInput{
+					Name:              scalingSpecDeploymentName,
+					Description:       utils.TestResourceDescription,
+					Region:            "us-east4",
+					CloudProvider:     "GCP",
+					Executor:          "CELERY",
+					SchedulerSize:     "SMALL",
+					IsDevelopmentMode: true,
+					ScalingSpec: `
+						scaling_spec = {
+							hibernation_spec = {
+								override = {
+								  is_hibernating = true
+								}
+							}
+						}`}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(scalingSpecResourceVar, "scaling_spec.hibernation_spec.override.is_hibernating", "true"),
+					resource.TestCheckNoResourceAttr(scalingSpecResourceVar, "scaling_spec.hibernation_spec.schedules"),
+				),
+			},
+			{
+				Config: astronomerprovider.ProviderConfig(t, true) + standardDeployment(standardDeploymentInput{
+					Name:              scalingSpecDeploymentName,
+					Description:       utils.TestResourceDescription,
+					Region:            "us-east4",
+					CloudProvider:     "GCP",
+					Executor:          "CELERY",
+					SchedulerSize:     "SMALL",
+					IsDevelopmentMode: true,
+					ScalingSpec: `
+						scaling_spec = {
+							hibernation_spec = {
+								override = {
+								  is_hibernating = true
+								  override_until = "2075-01-01T00:00:00Z"
+								}
+							}
+						}`}),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(scalingSpecResourceVar, "scaling_spec.hibernation_spec.override.is_hibernating", "true"),
+					resource.TestCheckResourceAttr(scalingSpecResourceVar, "scaling_spec.hibernation_spec.override.override_until", "2075-01-01T00:00:00Z"),
+					resource.TestCheckNoResourceAttr(scalingSpecResourceVar, "scaling_spec.hibernation_spec.schedules"),
```

Comment:
nit: for each config, can add comments in assertions on what we're checking - or maybe in the description field something like "Scaling Spec with Hibernation Spec Override Without Schedule" idk if it's overkill, but just an idea for test maintainability

Reply:
so I refactored the tests to make it a lil more obvious as to what they do
I think I will skip adding comments just in case we update the tests but forget to update the comments but now the tests should be a lot simpler and readable and the main gist of what is going on should be extremely obvious
Also the tests aren't necessarily for asserting the responses but more importantly to check that it doesnt error or panic when it should succeed